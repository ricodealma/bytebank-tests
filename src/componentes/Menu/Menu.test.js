import Menu from './index'
import {render, screen} from '@testing-library/react'

test('Deve renderizar um link para a página inicial', () => {
    render(<Menu/>);
    const linkPaginaInicial = screen.getByText('Inicial');
    expect(linkPaginaInicial).toBeInTheDocument();
});

test('Deve renderizar uma lista de links', () => {
    render(<Menu/>);
    const listaDeLinks = screen.getAllByRole('link');
    expect(listaDeLinks).toHaveLength(4);
});

test('Não deve renderizar um link para a página Extrato', () => {
    render(<Menu/>);
    const linkExtrato = screen.queryByText('Extrato');
    expect(linkExtrato).not.toBeInTheDocument();
});

test('Deve renderizar uma lista de links com a classe link', () => {
    render(<Menu/>);
    const links = screen.getAllByRole('link');
    links.forEach((link)=> {expect(link).toHaveClass('links');
    expect(link).toMatchSnapshot();
    });
    
})