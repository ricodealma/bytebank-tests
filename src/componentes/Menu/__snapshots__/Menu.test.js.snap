// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar uma lista de links com a classe link 1`] = `
<a
  class="links"
  href="/"
>
  Inicial
</a>
`;

exports[`Deve renderizar uma lista de links com a classe link 2`] = `
<a
  class="links"
  href="/"
>
  Transferências
</a>
`;

exports[`Deve renderizar uma lista de links com a classe link 3`] = `
<a
  class="links"
  href="/"
>
  Investimentos
</a>
`;

exports[`Deve renderizar uma lista de links com a classe link 4`] = `
<a
  class="links"
  href="/"
>
  Outros serviços
</a>
`;
